examples/demo-app/src/cloud-providers/aws/aws-provider.js

```diff
@@ -0,0 +1,316 @@
+
+      window.removeEventListener('message', handleLogin);
+
+      if (e.data.success) {
+        onCloudLoginSuccess();
+      }
+    };
+    window.addEventListener('message', handleLogin);
+  }
+
+  /**
+   * Required!
+   *
+   * @returns {Array<Viz>}
+   */
+  async listMaps() {
+    const s3PrivateFiles = await this._getMapListFromStorage('private');
+    const s3PublicFiles = await this._getMapListFromStorage('public');
+
+    return [...s3PublicFiles, ...s3PrivateFiles];
+  }
+
+  _getMapListFromStorage(level) {
+    return Storage.list('', {level})
+      .then(result => AwsProvider._prepareFileList(result, level))
+      .catch(e => {
+        const capitalizedLevel = level.charAt(0).toUpperCase() + level.slice(1);
+        AwsProvider._handleError(`${capitalizedLevel} maps failed to load`, e);
+      });
+  }
+
+  /**
+   * Generates array of viz objects from list of files from storage
+   * @returns {Array<Viz>}
+   */
+  static async _prepareFileList(fileList, level) {
+    const updatedFileList = [];
+    const mapFileList = fileList.filter(file => file.key.endsWith('map.json'));
+
+    for (const file of mapFileList) {
+      const title = file.key.split('.')[0];
+      const thumbnailKey = `${title}.thumbnail.png`;
+      const metaKey = `${title}.meta.json`;
+
+      const thumbnailURL = fileList.some(f => f.key === thumbnailKey)
+        ? await Storage.get(thumbnailKey, {
+            level,
+            download: false
+          }).catch(e => {
+            AwsProvider._handleError(`Map image ${thumbnailKey} failed to load`, e);
+          })
+        : null;
+
+      const description = fileList.some(f => f.key === metaKey)
+        ? await Storage.get(metaKey, {
+            level,
+            download: true
+          })
+            .then(metaFile => {
+              return metaFile.Body && metaFile.Body.description
+                ? metaFile.Body.description
+                : 'No description available.';
+            })
+            .catch(e => {
+              AwsProvider._handleError(`Description file ${thumbnailKey} failed to load`, e);
+            })
+        : 'No description available.';
+
+      updatedFileList.push({
+        id: file.key,
+        title,
+        description,
+        privateMap: level === 'private',
+        thumbnail: thumbnailURL,
+        lastModification: new Date(Date.parse(file.lastModified)),
+        loadParams: {
+          mapId: file.key,
+          privateMap: level === 'private',
+          level
+        }
+      });
+    }
+    return updatedFileList;
+  }
+
+  /**
+   * Required!
+   *
+   * @returns {MapResponse}
+   */
+  async downloadMap(loadParams) {
+    const {level, mapId, identityId} = loadParams;
+    const mapURL =
+      level === 'private'
+        ? await Storage.get(mapId, {level})
+        : await Storage.get(mapId, {level, identityId});
+
+    const mapData = await fetch(mapURL).then(response => response.json());
+    this._loadParam = loadParams;
```

Comment:
I don't quite understand what you're doing with the loadParm? Could you explain why you need to store them here on your instance